src/core/utils/helpers/general.ts

```diff
@@ -217,6 +217,42 @@ export const sortByReservationDate = (list: ReservationType[]) => {
   );
 };
 
+export const sortByDueDate = (list: LoanType[]) => {
+  //  We use orderBy from lodash to avoid mutating the original list
+  return orderBy(
+    list,
+    (item) =>
+      item.dueDate ? dayjs(item.dueDate).startOf("day").valueOf() : Infinity,
+    "asc"
+  );
+};
+
+export const sortLoansByIsRenewableThenDueDate = (list: LoanType[]) => {
+  //  We use orderBy from lodash to avoid mutating the original list
+  return orderBy(
+    list,
+    [
+      // First criterion: isRenewable (Use "desc" to put renewable items first)
+      (item) => Number(!!item.isRenewable),
+      // Second criterion: dueDate (earlier dates first)
+      (item) =>
+        item.dueDate ? dayjs(item.dueDate).startOf("day").valueOf() : Infinity
+    ],
+    ["desc", "asc"]
+  );
+};
```

Comment:
Consider just adding a single `sortByRenewable()` function and compose it with `sortByDueDate()`.

Personally I do not see the value in combining it. The combination also has some duplication from `sortByDueDate()`.